Source/player.h

```diff
@@ -29,14 +29,14 @@ enum {
 	// clang-format on
 };
 
-enum plr_class : uint8_t {
-	PC_WARRIOR,
-	PC_ROGUE,
-	PC_SORCERER,
-	PC_MONK,
-	PC_BARD,
-	PC_BARBARIAN,
-	NUM_CLASSES
+enum class HeroClass : uint8_t {
+	Warrior,
+	Rogue,
+	Sorcerer,
+	Monk,
+	Bard,
+	Barbarian,
+	ClassCount

```

Comment:
This doesn't have to happen in this PR, but it'd be nice if we moved the `ClassCount` member out of the num itself, e.g. into a `constexpr std::size_t kHeroClassCount = 7`

Having it within the enum means we can't benefit from the compiler's `switch` exhaustiveness check in switch statements.

One way to avoid having to manually specify the number is to define a `MAX = Barbarian` member and do `KHeroClassCount = HeroClass::MAX + 1`

Reply:
The external variable looks terrible, but the extra value inside the enum looks even worse so I agree. I actually thought about it but decided not to do it as part of this PR.

 > One way to avoid having to manually specify the number is to define a MAX = Barbarian member and do KHeroClassCount = HeroClass::MAX + 1

Adding a "MAX" entry, even if it is an alias, is also pretty bad IMHO. Suddenly, consumers could start using `MAX` instead of `Barbarian` which makes zero sense.

What about just defining it as:
```cpp
 kHeroClassCount = HeroClass::Barbarian + 1
```

BTW, what does that capital "K" in the name?

Reply:
this best live right next to the enum then so that another hero is added people will notice that this should be changed to  `kHeroClassCount = HeroClass::Newhero + 1`.

may i instead suggest that we get the size by counting the length of the arrays thathold the hero names, or similar arrays? https://github.com/diasurgical/devilutionX/blob/master/Source/player.cpp#L173

Reply:
> BTW, what does that capital "K" in the name?

I think some people prefix their constants with k. Like in "konstants". Because c was already taken to prefix classes. Hungarian notation...

Reply:
> Hungarian notation...

Pretty sure "konstant" is the danish word for it :thinking:

Reply:
> this best live right next to the enum then so that another hero is added people will notice that this should be changed to kHeroClassCount = HeroClass::Newhero + 1.
> 
> may i instead suggest that we get the size by counting the length of the arrays thathold the hero names, or similar arrays? https://github.com/diasurgical/devilutionX/blob/master/Source/player.cpp#L173

Hadn't seen this comment yet. As discussed, proposal for a more coupled approach (more coupled being better, in this case) is here : https://github.com/diasurgical/devilutionX/pull/1474/commits

Reply:
I guess that's way you where confused about it in the chat :)